tests/testthat/test-mplot.R

```diff
@@ -0,0 +1,15 @@
+test_that("maxample runs without error for all combinations of global and total", {
+  # Make sure no output is rendered
+  # Open a null graphics device (e.g., PDF without a file)
+  pdf(NULL)
+  # Ensure the device is closed after the test
+  on.exit(dev.off())
```

Comment:
```suggestion
  withr::defer(dev.off())
```
on.exit has some weird quirks, e.g. it is overwriting functions previously set to run on.exit (unless add = TRUE or so is passed). I think the documentation for withr::defer explains this more in depth. I try not to use on.exit at all.

Reply:
withr::local_pdf can probably do what you want here even more cleanly.